pages/Documents/ConversationalAI/Third-PartyBots/GoogleDialogflowES/advanced-features.md

```diff
@@ -379,13 +379,93 @@ After adding cloud function make sure to deploy it and verify it is active.
 
 After the function has been deployed this needs to be added to the fulfillment section of the Bot Configuration.
 This fulfillment can be found in the Google Dialogflow ES console as shown in the Figure 4.2 highlighted area.
-Webhook needs to be enabled and filled with the relevant information of the cloud function.
+Webhook need to be enabled and filled with the relevant information of the cloud function.
 (e.g. Auth Data and the Trigger URL)
 
 <img class="fancyimage" style="width:800px" src="img/dialogflowes/dialogflow_es_richcontentevent-cloud-function-data.png">
-Figure 4.2 Webhook configuration that need to be added for calling Cloud Function
+Figure 4.2 Webhook configuration that needs to be added for calling Cloud Function
 
 Once Webhook configuration is added then the Google Dialogflow ES bot will be able to respond to the requests via the cloud function.
 A demo of our WhatsApp map example with Google Cloud Function (defined above) can be seen below:
 
 <img class="fancyimage" style="width:300px" src="img/dialogflowes/dialog_flow_2_richcontent_demo.gif">
+
+### Receiving Last consumer message (Messaging Only)
+
+Third-Party bot now provides a way to add the last consumer message as a part of [the welcome event](third-party-bots-google-dialogflow-es-basic-content.html#the-welcome-event) (messaging only). When an ongoing conversation gets transferred to a new Agent or Skill, This enhancement will allow brands to respond to the last consumer message uttered as per their needs.
```

Comment:
you can remove `(messaging only)` from the description as it is in the heading already